components/raftstore/src/store/peer.rs

```diff
@@ -431,9 +432,13 @@ where
     /// of deleted entries.
     pub compaction_declined_bytes: u64,
     /// Approximate size of the region.
-    pub approximate_size: Option<u64>,
+    pub approximate_size: u64,
     /// Approximate keys of the region.
-    pub approximate_keys: Option<u64>,
+    pub approximate_keys: u64,
+    /// Whether this region has calculated region size by split-check thread. If we just splitted
+    ///  the region or ingested one file which may be overlapped with the existed data, the
+    /// `approximate_size` is not very accurate.
+    pub has_calculated_region_size: bool,
```

Comment:
It depends on the fact that `CasualMessage::RegionApproximateSize` msg is only sent by split checker and split check is done.

Reply:
No. It depends on that `CasualMessage::RegionApproximateSize` can give an accurate value.